pkg/collector/listeners/docker.go

```diff
@@ -73,17 +72,18 @@ func (l *DockerListener) Listen(newSvc chan<- Service, delSvc chan<- Service) {
 	l.newService = newSvc
 	l.delService = delSvc
 
-	// process containers that might be already running
-	l.init()
+CONNECT: // Outer loop handles re-subscribing
+	for {
+		messages, errs, err := l.dockerUtil.SubscribeToContainerEvents("DockerListener")
+		if err != nil {
+			log.Errorf("Cannot listen to docker events: %v", err)
+			signals.ErrorStopper <- true
```

Comment:
@hkaj do we still want to shut down the agent on AD errors, or leave that to the healthcheck?